facades/PC/src/main/java/org/terasology/engine/Terasology.java

```diff
@@ -150,18 +153,30 @@ public static void main(String[] args) {
             if (isHeadless) {
                 engine.subscribeToStateChange(new HeadlessStateChangeListener(engine));
                 engine.run(new StateHeadlessSetup());
+            } else if (loadLastGame) {
+                engine.getFromEngineContext(ThreadManager.class).submitTask("loadGame", () -> {
+                    GameManifest gameManifest = getLatestGameManifest();
+                    if (gameManifest != null) {
+                        engine.changeState(new StateLoading(gameManifest, NetworkMode.NONE));
+                    }
+                });
             } else {
-                if (loadLastGame) {
-                    engine.getFromEngineContext(ThreadManager.class).submitTask("loadGame", () -> {
+                if (createLastGame) {
+                    engine.initialize();
+                    engine.getFromEngineContext(ThreadManager.class).submitTask("createLastGame", () -> {
                         GameManifest gameManifest = getLatestGameManifest();
                         if (gameManifest != null) {
-                            engine.changeState(new StateLoading(gameManifest, NetworkMode.NONE));
+                            GameManifest newlyCreatedGM = gameManifest;
+                            Random random = new Random();
+                            String randomNumber = String.valueOf(random.nextInt());
+                            newlyCreatedGM.setTitle("New Created " + gameManifest.getSeed() + "(" + randomNumber + ")");
+                            engine.changeState(new StateLoading(newlyCreatedGM, NetworkMode.NONE));
```

Comment:
Setting the title to something else will rebuild the world with the same seed.

Reply:
This overall looks reasonable, but I wonder if we could name the game more cleanly - if you've got the previous game manifest does that by chance include that save's _name_ and if so could we just do "Recreated '[old game name]'" or so? Maybe with an incrementing number rather than a random one?

Just recently https://github.com/MovingBlocks/Terasology/pull/4064 was merged which was trying to harden the choice of which increment would get added at the end of a game name, including _clever_ cases where a user tries to trick the system (by for instance adding an increment manually that would clash with something). Mind taking a look at that and seeing if we have any ideal options? :)